symbol/symbol_store.go

```diff
@@ -19,25 +19,35 @@ import (
 	"encoding/hex"
 	"errors"
 	"io"
+	"io/ioutil"
+	"os"
+	"path"
 
 	"github.com/go-kit/kit/log"
 	"github.com/go-kit/kit/log/level"
 	"github.com/gogo/status"
 	"github.com/thanos-io/thanos/pkg/objstore"
 	"google.golang.org/grpc/codes"
 
+	"github.com/conprof/conprof/internal/pprof/binutils"
 	"github.com/conprof/conprof/pkg/store/storepb"
 )
 
 type SymbolStore struct {
-	bucket objstore.Bucket
-	logger log.Logger
+	bucket   objstore.Bucket
+	logger   log.Logger
+	bu       *binutils.Binutils
+	cacheDir string
 }
 
-func NewSymbolStore(logger log.Logger, bucket objstore.Bucket) *SymbolStore {
+func NewSymbolStore(logger log.Logger, bucket objstore.Bucket, cacheDir string) *SymbolStore {
+	bu := &binutils.Binutils{}
+	level.Debug(logger).Log("msg", "initializing binutils", "binutils", bu.String())
```

Comment:
I am not familiar with `binutils`, but what's the use case of printing `bu.String()`. It has not been initialized yet.

Reply:
it's a little convoluted but `bu.String()` initializes the binutils object if it hasn't already been .. basically this prints the external tools that were discovered and will be used .. I left this here because I had trouble understanding the difference in environments from my local machine and github actions, and this is how I found it, so I felt it was useful to keep for future troubleshooting.